plugins/GSdx/GSTextureCache.h

```diff
@@ -105,6 +106,55 @@ class GSTextureCache
 		void Update();
 	};
 
+	class Palette
+	{
+	private:
+		uint16 m_ref_counter = 0; // Reference counter
+		uint32* m_clut; // Pointer to a copy of relevant clut
+		GSTexture* m_tex_palette; // Pointer to valid texture with relevant clut as content
+
+	public:
+		Palette(); // Default constructor
+		Palette(const Palette& palette); // Copy constructor
+		Palette(const GSRenderer* renderer, uint16 pal); // Creates a copy of the current clut and a texture with its content
+
+		// Getter for clut pointer
+		__forceinline uint32* GetClut();
+
+		// Getter for palette texture pointer
+		__forceinline GSTexture* GetPaletteGSTexture();
+
+		// Getter for the reference counter
+		__forceinline uint16 GetRefCounter();
```

Comment:
It feels like it would be better to use `std::shared_ptr<Palette>` pointer to manage the reference counting.

Reply:
Well the scope of usage is limited, for now. I don't feel confident with `std::shared_ptr` yet, but it may help avoid some extra lines of code.